src/CodeGeneration/ApiGenerator/Generator/ApiEndpointFactory.cs

```diff
@@ -102,5 +104,76 @@ void ReplaceOptions(string path)
 			ReplaceOptions("*.url.parts.metric.options");
 			ReplaceOptions("*.url.parts.index_metric.options");
 		}
+
+		/// <summary>
+		/// Changes the structure of new REST API spec in 7.4.0 to one that matches prior spec structure.
+		/// </summary>
+		private static void TransformNewSpecStructureToOld(JObject original)
```

Comment:
I think this is the right approach on our 7.x branch, in master we might need to look in to moving our in memory representation over as well. This can be forward ported to master in the mean time.

Love the pragmatic solution though :+1: